src/main/java/com/karankumar/bookproject/ui/book/BookForm.java

```diff
@@ -355,9 +358,11 @@ private boolean isMovedToDifferentShelf() {
     }
 
     private void setBookBean() {
-        Book book = populateBookBean();
-        if (book != null) {
-            binder.setBean(book);
+
+
+        Optional<Book> book = populateBookBean();
+        if (book.isPresent()) {
+            binder.setBean(book.get());
```

Comment:
Can be replaced with:
`populateBookBean().ifPresent(value -> binder.setBean(value));`